src/NLog.Web.AspNetCore/Config/SetupBuilderExtensions.cs

```diff
@@ -25,7 +25,12 @@ public static ISetupBuilder LoadConfigurationFromAppSettings(this ISetupBuilder
             var currentBasePath = basePath;
             if (currentBasePath is null)
             {
-                currentBasePath = AppContext.BaseDirectory;
+                var webHostDefaultConfig = new ConfigurationBuilder()
+                    .AddEnvironmentVariables(prefix: "ASPNETCORE_")
+                    .Build();
+
+                var contentRootPath = webHostDefaultConfig.GetValue<string>(Microsoft.AspNetCore.Hosting.WebHostDefaults.ContentRootKey);
+                currentBasePath = ResolveContentRootPath(contentRootPath, AppContext.BaseDirectory);
```

Comment:
What happens if you use this instead:
```c#
if (currentBasePath is null)
{
    currentBasePath = currentBasePath ?? GetAspNetCoreEnvironment("ASPNETCORE_CONTENTROOT") ?? GetAspNetCoreEnvironment("DOTNET_CONTENTROOT") ?? AppContext.BaseDirectory;
    if (string.IsNullOrEmpty(currentBasePath))
    {
       currentBasePath = Directory.GetCurrentDirectory();
    }
}
```

Reply:
The environment variables for `ASPNETCORE_CONTENTROOT` and `DOTNET_CONTENTROOT` are not always set. I tested locally with an ASP.Net Core project of mine.

Reply:
It's true that the environment variables should override the configuration though. I can update the PR with that in mind.

Reply:
Tried calling this method:
```
currentBasePath = new ConfigurationBuilder().AddEnvironmentVariables(prefix: "ASPNETCORE_").Build().GetValue<string>(Microsoft.AspNetCore.Hosting.WebHostDefaults.ContentRootKey);
``
And can confirm that your method is also not working

Reply:
Sorry for this inconvenience. I'll try and find an alternative solution in another PR.

Reply:
But I do like including environment-variable `ASPNETCORE_CONTENTROOT`, so will include that.